src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java

```diff
@@ -129,21 +138,43 @@ public PasswordResetLinkServiceImpl(ConfigurationClient configurationClient, Aut
         String generatedLink = linkHost + linkPath + '/' + token;
         linkHolder.value = generatedLink;
 
+        String expirationTimeFromConfig =
+          configMapHolder.value.getOrDefault(LINK_EXPIRATION_TIME_CONFIG_KEY, LINK_EXPIRATION_TIME_DEFAULT);
+        String expirationUnitOfTimeFromConfig = configMapHolder.value.getOrDefault(
+          LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, LINK_EXPIRATION_UNIT_OF_TIME_DEFAULT);
+
         String eventConfigName = passwordExistsHolder.value ? RESET_PASSWORD_EVENT_CONFIG_NAME : CREATE_PASSWORD_EVENT_CONFIG_NAME;
         Notification notification = new Notification()
           .withEventConfigName(eventConfigName)
           .withRecipientId(userId)
           .withContext(
             new Context()
               .withAdditionalProperty("user", JsonObject.mapFrom(userHolder.value))
-              .withAdditionalProperty("link", generatedLink))
+              .withAdditionalProperty("link", generatedLink)
+              .withAdditionalProperty("expirationTime", expirationTimeFromConfig)
+              .withAdditionalProperty("expirationUnitOfTime", expirationUnitOfTimeFromConfig))
           .withText(StringUtils.EMPTY)
           .withLang(DEFAULT_NOTIFICATION_LANG);
         return notificationClient.sendNotification(notification, connectionParams);
       })
       .map(v -> linkHolder.value);
   }
 
+  private long convertDateToMilliseconds(String expirationTimeDefault, String expirationUnitOfTime) {
+    long expirationTime = Long.parseLong(expirationTimeDefault);
```

Comment:
Shouldn't we handle NumberFormatException? If I understand correctly, this value may come from mod-config, so it's dependent on user's input.

Reply:
Good point, I added logic which will return default value

Reply:
Now it looks like we treat two similar cases differently. If expiration time is invalid we use the default value instead, but in case of invalid unit we throw an exception. Could we throw an exception in both cases?

Reply:
Changed

Reply:
Thanks!